misc/lib-creator/src/main/java/org/ballerinalang/stdlib/utils/GenerateBalo.java

```diff
@@ -159,23 +168,23 @@ private static void genApiDoc(String sourceRootDir, Set<String> docModuleFilter,
         BallerinaDocGenerator.writeAPIDocsForModules(moduleDocMap, apiDocPath, false);
     }
 
-    private static void printErrors(boolean reportWarnings, CompileResult.CompileResultDiagnosticListener diagListner,
+    private static void printErrors(boolean reportWarnings,
                                     List<Diagnostic> diagnostics, boolean logIsolationWarnings) {
         int deprecatedWarnCount = 0;
-        if (reportWarnings && diagListner.getWarnCount() > 0) {
-            for (Diagnostic diagnostic : diagListner.getDiagnostics()) {
-                DiagnosticCode code = diagnostic.getCode();
+        if (reportWarnings && warnCount > 0) {
+            for (Diagnostic diagnostic : diagnostics) {
+                DiagnosticCode code = ((BLangDiagnostic) diagnostic).getCode();
                 if (code == USAGE_OF_DEPRECATED_CONSTRUCT || (!logIsolationWarnings && isIsolatedWarningLog(code))) {
                     deprecatedWarnCount++;
                 }
             }
         }
-        if (diagListner.getErrorCount() > 0 ||
-                (reportWarnings && (diagListner.getWarnCount() - deprecatedWarnCount) > 0)) {
+        if (errorCount > 0 ||
+                (reportWarnings && (warnCount - deprecatedWarnCount) > 0)) {
             StringJoiner sj = new StringJoiner("\n  ");
             diagnostics.forEach(e -> sj.add(e.toString()));
```

Comment:
We shouldn't print the diags here. Since we are already printing them at the `Compiler` class, this will result it duplicate errors

Reply:
So, are we removing L185 ?

Reply:
yes, we can remove that.

Reply:
okay